### PR TITLE
Optimize lookup match rules

### PR DESCRIPTION
**Pull request by cr7258**
Optimize match rules of the lookup command as follows:
- If the `-e` parameter is not specified, return service accounts that match exactly.
- If the `-e` parameter is specified, return service accounts that match regex.
- If no service account is provided for searching, return all service accounts.

For example:
```bash
rbac-tool lookup "kubelet"

# exact match
  SUBJECT | SUBJECT TYPE | SCOPE       | NAMESPACE | ROLE                                   | BINDING                                 
+---------+--------------+-------------+-----------+----------------------------------------+----------------------------------------+
  kubelet | User         | ClusterRole |           | system:node-problem-detector           | kubelet-user-npd-binding                
  kubelet | User         | ClusterRole |           | system:node-bootstrapper               | kubelet-bootstrap                       
  kubelet | User         | ClusterRole |           | gce:beta:kubelet-certificate-bootstrap | gce:beta:kubelet-certificate-bootstrap  
```

```bash
rbac-tool lookup -e "kubelet"

# regex match
  SUBJECT           | SUBJECT TYPE | SCOPE       | NAMESPACE | ROLE                                   | BINDING                                  
+-------------------+--------------+-------------+-----------+----------------------------------------+-----------------------------------------+
  kubelet           | User         | ClusterRole |           | gce:beta:kubelet-certificate-bootstrap | gce:beta:kubelet-certificate-bootstrap   
  kubelet           | User         | ClusterRole |           | system:node-bootstrapper               | kubelet-bootstrap                        
  kubelet           | User         | ClusterRole |           | system:node-problem-detector           | kubelet-user-npd-binding                 
  kubelet-bootstrap | User         | ClusterRole |           | gce:beta:kubelet-certificate-bootstrap | kubelet-bootstrap-certificate-bootstrap  
  kubelet-bootstrap | User         | ClusterRole |           | system:node-bootstrapper               | kubelet-bootstrap-node-bootstrapper  
```

```bash
rbac-tool lookup

# return all service accounts
 SUBJECT                               | SUBJECT TYPE   | SCOPE       | NAMESPACE                | ROLE                                                 | BINDING                                               
+---------------------------------------+----------------+-------------+--------------------------+------------------------------------------------------+------------------------------------------------------+
  antrea-controller                     | ServiceAccount | ClusterRole |                          | antrea-controller                                    | antrea-controller                                     
  antrea-cpha                           | ServiceAccount | ClusterRole |                          | antrea-cpha                                          | antrea-cpha                                           
  antrea-cpha                           | ServiceAccount | Role        | kube-system              | antrea-cpha                                          | antrea-cpha                                           
  attachdetach-controller               | ServiceAccount | ClusterRole |                          | system:controller:attachdetach-controller            | system:controller:attachdetach-controller             
  bootstrap-signer                      | ServiceAccount | Role        | kube-public              | system:controller:bootstrap-signer                   | system:controller:bootstrap-signer                    
  bootstrap-signer                      | ServiceAccount | Role        | kube-system              | system:controller:bootstrap-signer                   | system:controller:bootstrap-signer                    
  cert-controller-manager               | ServiceAccount | ClusterRole |                          | cert-controller-manager                              | cert-controller-manager                               
  cert-controller-manager               | ServiceAccount | Role        | gardener                 | cert-controller-manager                              | cert-controller-manager                               
  certificate-controller                | ServiceAccount | ClusterRole |                          | system:controller:certificate-controller             | system:controller:certificate-controller       
......
```

### CODE REVIEW

**Review by gadinaor-r7 — APPROVED** [review]
LGTM - thanks @cr7258 !!!

File: cmd/lookup_cmd.go [hunk 1]
```diff
@@ -47,14 +50,17 @@ rbac-tool lookup -ne '^system:.*'
 			var re *regexp.Regexp
 			var err error
 
-			if regex != "" {
-				re, err = regexp.Compile(regex)
-			} else {
-				if len(args) != 1 {
-					re, err = regexp.Compile(fmt.Sprintf(`.*`))
+			if regex == "" {
+				if len(args) > 0 {
```
Comment (gadinaor-r7) [inline cmd/lookup_cmd.go:54]: args should be exactly one

Comment (cr7258) [reply cmd/lookup_cmd.go]: Done: https://github.com/alcideio/rbac-tool/pull/98/commits/06bc650f42c377ff5fdfe27a4a23af3efb7edb7f

**Review by gadinaor-r7 — APPROVED** [review]
LGTM

**Pull request merged**